ipa/ipa_provider.py

```diff
@@ -475,8 +475,9 @@ def test_image(self):
                     with open(self.log_file, 'a') as log_file:
                         with ipa_utils.redirect_output(log_file):
                             # Run tests
-                            status = (status or
-                                      self._run_tests(item, ssh_config))
+                            result = self._run_tests(item, ssh_config)
+                            status = status or result
```

Comment:
status is never assigned thus this is really "status = 0 or result" which always amounts to result unless of course result is also 0 in which case the shortcut algorithm would use the 0 from "status" but that's semantics, thus this should be status = result, skip the superfluous branch.

Reply:
Not sure I follow:

- Status is defaulted to 0 for successful exit.
- Every pytest invocation returns the exit code.
- If any pytest invocation returns a none zero exit code then we want to set that to status and maintain that value.
  - Unless `--early-exit` has been provided we want to run all tests even if there's a failure

With the current implementation if there are multiple pytest runs and one fails then no other tests are run which is not what we want.

And we can't throw away a none zero exit code if there are more tests. Otherwise the overall test status is invalid.

E.g. three different groups of tests are run with exit codes in order: 0, 1, 0. The status should be 1 since there was a failure.

Reply:
Never mind I missed the loop